### PR TITLE
Stop preload while prewarmup is running

### PR DESCRIPTION
**Pull request by engahmeds3ed**
## Description

We found when preload is enabled and the RUCSS option is enabled, that it's a big headache on the client's server so we need to delay preload till prewarmup finishes its work.

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Manually enable RUCSS when preload is working => should stop preload and start prewarmup
- [ ] Manually enable RUCSS when preload is enabled but not working => should start prewarmup and when finished the preload will start.
- [ ] Manually enable RUCSS while preload is disabled => should start prewarmup only
- [ ] When preload is enabled and manually clicked the button for forcing preload from the top menu or settings dashboard while prewarmup is working => prewarmup will continue and nothing will happen with preload.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: inc/API/preload.php [hunk 1]
```diff
@@ -91,6 +91,12 @@ function do_admin_post_rocket_preload_cache() { // phpcs:ignore WordPress.Naming
 		die();
 	}
 
+	$prewarmup_stats = get_option( 'wp_rocket_prewarmup_stats' );
+	if ( empty( $prewarmup_stats['allow_optimization'] ) ) {
```
Comment (remyperona) [inline inc/API/preload.php]: We also need to check for the value of `remove_unused_css` here, else we would bail out all the time, even if the option is not enabled

Comment (engahmeds3ed) [reply inc/API/preload.php]: Done

**Comment by remyperona** [comment]
Can you update the checklist items, notably the `How Has This Been Tested?` section, to detail how you have tested this change?

**Comment by engahmeds3ed** [comment]
I will do that tomorrow morning after adjusting the method `maybe_cancel_preload` to also cancel preload when RUCC option is enabled so if preload was working that should stop it.

**Review by remyperona — APPROVED** [review]

**Comment by piotrbak** [comment]
@engahmeds3ed I think we're good here. Please confirm that the below testing is enough:
1. Prewarmup working + Preload triggered from the top menu = Preload not activated
2. Prewarmup working +  Activate the preload (it was disabled before) = Preload not activated
3. Preload working + Activate RUCSS = Preload is stopped and Prewarmup starts
4. Preload enabled but not working + Activate RUCSS = Prewarmup starts

After each scenario, the Preload is triggered.
[testrail-report-182.pdf](https://github.com/wp-media/wp-rocket/files/6524510/testrail-report-182.pdf)

**Comment by engahmeds3ed** [comment]
Thanks @piotrbak 
Yes, I think this is enough, just a small thing:
Preload also can be manually triggered from the dashboard button

image

so we need to make sure that point number 1 is valid with this button also.
(Not a big deal, both are sharing the same functionality)

Thanks.

**Review by piotrbak — APPROVED** [review]

**Pull request merged**